nixos/modules/services/mail/postal.nix

```diff
@@ -0,0 +1,926 @@
+          default = "OIDC Provider";
+          description = "The name of the OIDC provider as shown in the UI";
+        };
+
+        issuer = mkOption {
+          type = with types; nullOr str;
+          default = null;
+          description = "The OIDC issuer URL";
+        };
+
+        identifier = mkOption {
+          type = with types; nullOr str;
+          default = null;
+          description = "The client ID for OIDC";
+        };
+
+        scopes = mkOption {
+          type = with types; listOf str;
+          default = [
+            "openid"
+            "email"
+          ];
+          description = "Scopes to request from the OIDC server.";
+        };
+
+        uid_field = mkOption {
+          type = types.str;
+          default = "sub";
+          description = "The field to use to determine the user's UID";
+        };
+
+        email_address_field = mkOption {
+          type = types.str;
+          default = "email";
+          description = "The field to use to determine the user's email address";
+        };
+
+        name_field = mkOption {
+          type = types.str;
+          default = "name";
+          description = "The field to use to determine the user's name";
+        };
+
+        discovery = mkOption {
+          type = types.bool;
+          default = true;
+          description = "Enable discovery to determine endpoints from .well-known/openid-configuration from the Issuer";
+        };
+
+        authorization_endpoint = mkOption {
+          type = with types; nullOr str;
+          default = null;
+          description = "The authorize endpoint on the authorization server (only used when discovery is false)";
+        };
+
+        token_endpoint = mkOption {
+          type = with types; nullOr str;
+          default = null;
+          description = "The token endpoint on the authorization server (only used when discovery is false)";
+        };
+
+        userinfo_endpoint = mkOption {
+          type = with types; nullOr str;
+          default = null;
+          description = "The user info endpoint on the authorization server (only used when discovery is false)";
+        };
+
+        jwks_uri = mkOption {
+          type = with types; nullOr str;
+          default = null;
+          description = "The JWKS endpoint on the authorization server (only used when discovery is false)";
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = optionals cfg.config.smtp_server.tls_enabled [
+      {
+        assertion = cfg.config.smtp_server.tls_certificate_path != null;
+        message = "`services.postal.config.smtp_server.tls_certificate_path` must be set when TLS is enabled.";
+      }
+      {
+        assertion = cfg.config.smtp_server.tls_private_key_path != null;
+        message = "`services.postal.config.smtp_server.tls_private_key_path` must be set when TLS is enabled.";
+      }
+    ];
+
+    systemd.tmpfiles.rules = [
```

Comment:
Maybe checkout `systemd.tmpfiles.settings` for an attribute based interface for tmpfiles ^^

Reply:
Done :)